torque-core/src/main/kotlin/com/workday/torque/ChunkRetryer.kt

```diff
@@ -0,0 +1,76 @@
+package com.workday.torque
+
+import com.linkedin.dex.parser.TestMethod
+import com.workday.torque.pooling.TestChunk
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.withTimeout
+import java.util.concurrent.TimeUnit
+
+class ChunkRetryer(private val adbDevice: AdbDevice,
+                   private val args: Args,
+                   private val logcatFileIO: LogcatFileIO,
+                   private val testChunkRunner: TestChunkRunner,
+                   private val installer: Installer
+) {
```

Comment:
What do you think about having an interface that TestChunkRunner and ChunkRetryer both implement, since the `runTestChunkWithRetry` method signatue is basically the same as `run`, and you can just pass in the `args` arg as a method arg instead of a constructor arg? And then if we want additional functionality in the future, it's easy to add it on with additional composition and inject that new instance.

Reply:
So my thought is that TestChunkRetryer does things that crosses retries, and the chunkRunner would contain things that applies to every single run. So the retryer wouldn't be on the same logical layer as the chunkRunner.
So for example the TestChunkRetryer would probably have cross-retries-chunk-run `VideoRecordPlugin` of a CrossRetryPlugin interface, and TestChunRunner would have `LogcatPlugin` of a PerRunPlugin interface.
Any I think it'd make more sense when we actually think of additional requirements?